tests/elster_xml/grundsteuer/test_elster_data_representation.py

```diff
@@ -7,12 +7,256 @@
+        gebaeude = SampleGebaeude().with_wohnflaeche(10).with_weitere_wohnraeume(42, 2).parse()
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Weitere_Wohn is not None
+
+    def test_on_weitere_wohnraeume_flag_false_should_not_set_weitere_wohnraeume(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(10).with_weitere_wohnraeume(42, 2).parse()
+        gebaeude.weitere_wohnraeume.has_weitere_wohnraeume = False
+
+        result = EAngDurchschn(gebaeude)
+
+        assert result.Weitere_Wohn is None
+
+
+class TestEAngWohn:
+    def test_on_ab_1949_should_contain_baujahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_baujahr("1959").parse()
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403114 == "1959"
+
+    def test_on_vor_1949_should_not_contain_baujahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_baujahr("1959").parse()
+        gebaeude.ab1949.is_ab1949 = False
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403114 is None
+
+    def test_on_kenrsaniert_should_contain_kernsanierungsjahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_kernsanierung("1959").parse()
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403115 == "1959"
+
+    def test_on_not_kenrsaniert_should_not_contain_kernsanierungsjahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_kernsanierung("1959").parse()
+        gebaeude.kernsaniert.is_kernsaniert = False
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403115 is None
+
+    def test_on_abbruchverpflichtung_should_contain_abbruchverpflichtungsjahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_abbruchverpflichtung("1959").parse()
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403116 == "1959"
+
+    def test_on_no_abbruchverpflichtung_should_not_contain_abbruchverpflichtungsjahr(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_abbruchverpflichtung("1959").parse()
+        gebaeude.abbruchverpflichtung.has_abbruchverpflichtung = False
+
+        result = EAngWohn(gebaeude)
+
+        assert result.E7403116 is None
+
+    def test_on_garagen_should_contain_set_anzahl(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_garagen(2).parse()
+
+        result = EAngWohn(gebaeude)
+
+        assert result.Garagen.E7403171 == 2
+
+    def test_on_no_garagen_should_contain_not_contain_garagen(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).with_garagen(2).parse()
+        gebaeude.garagen.has_garagen = False
+
+        result = EAngWohn(gebaeude)
+
+        assert result.Garagen is None
+
+    def test_should_set_ang_durchschn(self):
+        gebaeude = SampleGebaeude().with_wohnflaeche(42).parse()
+
+        result = EAngWohn(gebaeude)
+
+        assert result.Ang_Durchschn is not None
```

Comment:
You know hwat I wanna say. :D

Reply:
That we're not testing the instance's state? See my comments above... unless I misunderstood.